src/org/skypro/skyshop/searchengine/SearchEngine.java

```diff
@@ -13,11 +15,11 @@ public SearchEngine() {
         searchlist = new ArrayList<>();
     }
 
-    public List<Searchable> search(String target) {
-        List<Searchable> searchResult = new ArrayList<>();
+    public SortedMap<String, Searchable> search(String target) {
```

Comment:
```suggestion
    public Map<String, Searchable> search(String target) {
```
не забываем про полиморфизм